_rules/explicit-SVG-image-has-name-7d6734.md

```diff
@@ -0,0 +1,187 @@
+---
+id: 7d6734
+name: '`svg` element with explicit role has accessible name'
+rule_type: atomic
+description: |
+  This rule checks that each SVG image element that is explicitly included in the accessibility tree has an accessible name.
+accessibility_requirements:
+  wcag20:1.1.1: # Non-Text Content (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+authors:
+  - Bryn Anderson
+---
+
+## Applicability
+
+The rule applies to any element in the [SVG](https://www.w3.org/2000/svg) namespace with an [explicit semantic role](#explicit-semantic-role) of either `img`, `graphics-document`, `graphics-symbol`, that is [included in the accessibility tree](#included-in-the-accessibility-tree).
+
+**Note**: The [SVG Accessibility API Mappings](https://www.w3.org/TR/svg-aam-1.0/#include_elements) specifies that many elements in the SVG namespace are purely presentational and should not be included in the accessibility tree unless indicated otherwise through the use of text alternative content, an explicit WAI ARIA role, or a valid `tabindex` attribute.
+
+## Expectation
+
+Each target element has an [accessible name](#accessible-name) that is not only [whitespace](#whitespace).
```

Comment:
Note: as per #976 it seems that every element has a (possibly empty) accessible name and that accessible name are never (or should never be) only whitespace (_i.e._ every applicable element should pass the rule).

Given that this is a general problem, I guess we should proceed with this rule and solve the problem once we have discussed the issue and reached a decision.

Reply:
This needs to be addressed. Given that issue now has a PR that's in final call, this needs to be fixed before merging. Change "`an [accessible name][] that is not only [whitespace][]`" to "`an [accessible name][] is empty`."

I think it's reasonable to do this as an editorial change and not have it go through another final call.

Reply:
Updated as per Passed Example 4 in https://github.com/act-rules/act-rules.github.io/pull/981/files#diff-e5855eef878bfc5e2ac47f8ae6846dc4